bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCHandler.java

```diff
@@ -128,18 +148,48 @@ public void processUpdate(String serviceName, JsonElement stateData) {
         return (BoschSHCBridgeHandler) bridge.getHandler();
     }
 
+    /**
+     * Creates and registers a new service for this device.
+     * 
+     * @param <TService> Type of service.
+     * @param <TState> Type of service state.
+     * @param serviceClass Class of service to instantiate a new instance.
+     * @param stateUpdateListener Function to call when a state update was received
+     *            from the device.
+     * @param affectedChannels Channels which are affected by the state of this
+     *            service.
+     * @return Instance of registered service.
+     */
+    protected <TService extends BoschSHCService<TState>, TState extends BoschSHCServiceState> TService createService(
+            Class<TService> serviceClass, Consumer<TState> stateUpdateListener, Collection<String> affectedChannels) {
+        BoschSHCBridgeHandler bridgeHandler = this.getBridgeHandler();
+        if (bridgeHandler == null) {
+            throw new Error(String.format("Could not initialize service for {}, no valid bridge set", this.getThing()));
+        }
+
+        String deviceId = this.getBoschID();
+
+        TService service;
+        try {
+            service = serviceClass.newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new Error(String.format("Could not create instance of service {}", serviceClass.getName()));
+        }
+        service.initialize(bridgeHandler, deviceId, stateUpdateListener);
+        this.registerService(service, affectedChannels);
+
+        return service;
+    }
+
     /**
      * Registers a service of this device.
      * 
-     * @param service          Service which belongs to this device
+     * @param service Service which belongs to this device
      * @param affectedChannels Channels which are affected by the state of this
-     *                         service
+     *            service
      */
     protected <TState extends BoschSHCServiceState> void registerService(BoschSHCService<TState> service,
```

Comment:
You should change the function to private if ``createService`` is the new default way, else new handler code could use the old function by mistake.
```suggestion
    private <TState extends BoschSHCServiceState> void registerService(BoschSHCService<TState> service,
```

Reply:
Yes, my idea was to make it the new default way. The registerService method was still in use by the ThermostatHandler, so I left it as protected for now. But you are right, I can refactor it directly to avoid confusion.